### PR TITLE
feat(runtime)!: Removes old guest config and uses runtime config instead

### PR DESCRIPTION
**Pull request by thomastaylor312**
Most of the changes are related to wit updates, but this removes the guest config from `wasmcloud:bus` and pulls down `wasi:config` in its place

### CODE REVIEW

File: crates/runtime/src/actor/component/mod.rs [hunk 1]
```diff
@@ -465,8 +467,7 @@ fn instantiate(
             | "wasi:io/poll@0.2.0"
             | "wasi:io/streams@0.2.0"
             | "wasi:sockets/tcp@0.2.0"
-            | "wasmcloud:bus/lattice"
-            | "wasmcloud:bus/guest-config" => continue,
```
Comment (brooksmtownsend) [inline crates/runtime/src/actor/component/mod.rs:469]: :excellent:

File: crates/runtime/src/actor/component/config.rs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    }
+}
+
+#[async_trait]
+impl runtime::Host for Ctx {
+    #[instrument(skip(self))]
+    async fn get(
+        &mut self,
+        key: String,
+    ) -> anyhow::Result<Result<Option<String>, config::runtime::ConfigError>> {
```
Comment (brooksmtownsend) [inline crates/runtime/src/actor/component/config.rs:25]: I know the runtime interface is implemented as a `Result<Option<String>>`, is there a reason why we double wrap this here with another anyhow Result?

**Review by brooksmtownsend — APPROVED** [review]
LGTM and approved even though I had a question because I see we use that pattern elsewhere

File: crates/runtime/src/actor/component/config.rs (hunk 2 above)
Comment (thomastaylor312) [reply crates/runtime/src/actor/component/config.rs:25]: The result is something defined by the host bindings insofar as I understand. I think everything is wrapped by a result type

**Pull request merged**